src/mscorlib/shared/System/IO/PathInternal.cs

```diff
@@ -140,5 +140,20 @@ internal static void CheckSearchPattern(string searchPattern)
                 searchPattern = searchPattern.Substring(index + 2);
             }
         }
+
+        internal static string NormalizeDirectorySeparators(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+                return path;
+
+            string result = NormalizeDirectorySeparatorsIfNeccessary(path);
+            return result != null ? result : path;
```

Comment:
Why not use string.IsNotNullOrEmpty here?

Reply:
result can't be empty if path is string. Null is returned if the path is already normalized

Reply:
then you can do:

```cs
return NormalizeDirectorySeparatorsIfNeccessary(path) ?? path;
```

if it is null, it will return path.

Reply:
You could also do:

```cs
internal static string NormalizeDirectorySeparators(string path) => string.IsNullOrEmpty(path) ? path : NormalizeDirectorySeparatorsIfNeccessary(path) ?? path;
```